openvdb_houdini/openvdb_houdini/SOP_OpenVDB_Merge.cc

```diff
@@ -0,0 +1,682 @@
+        fpreal mTime = 0.0;
+    };
+};
+
+
+void
+newSopOperator(OP_OperatorTable* table)
+{
+    if (table == nullptr) return;
+
+    hutil::ParmList parms;
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "group", "Group")
+        .setTooltip("Specify a subset of the input VDBs to be modified.")
+        .setChoiceList(&hutil::PrimGroupMenuInput1)
+        .setDocumentation(
+            "A subset of the input VDBs to be modified"
+            " (see [specifying volumes|/model/volumes#group])"));
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "collation", "Collation")
+        .setDefault("nameclassandtype")
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "nameclassandtype", "Name, VDB Class and Value Type",
+            "classandtype",     "VDB Class and Value Type"
+        })
+        .setTooltip(
+            "Criteria under which groups of grids are merged.")
+        .setDocumentation(
+            "The criteria under which groups of grids are merged.\n\n"
+            "Grids with different names will be merged if name is not part"
+            " of the collation criteria"));
+
+    // Menu of resampling options
+    parms.add(hutil::ParmFactory(PRM_STRING, "resample", "Resample VDBs")
+        .setDefault("first")
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "first",        "To Match First VDB"
+        })
+        .setTypeExtended(PRM_TYPE_JOIN_PAIR)
+        .setDocumentation("Specify which grid to use as reference when resampling."));
+
+    // Menu of resampling interpolation order options
+    parms.add(hutil::ParmFactory(PRM_ORD, "resampleinterp", "Interpolation")
+        .setDefault(PRMoneDefaults)
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "point",     "Nearest",
+            "linear",    "Linear",
+            "quadratic", "Quadratic"
+        })
+        .setTooltip(
+            "Specify the type of interpolation to be used when\n"
+            "resampling one VDB to match the other's transform.")
+        .setDocumentation(
+            "The type of interpolation to be used when resampling one VDB"
+            " to match the other's transform\n\n"
+            "Nearest neighbor interpolation is fast but can introduce noticeable"
+            " sampling artifacts.  Quadratic interpolation is slow but high-quality."
+            " Linear interpolation is intermediate in speed and quality."));
+
+    parms.beginSwitcher("Group1");
+
+    parms.addFolder("Merge Operation");
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "op_fog", "Fog VDBs")
+        .setDefault("add")
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "none",             "None            ",
+            "add",              "Add"
+        })
+        .setTooltip("Merge operation for Fog VDBs.")
+        .setDocumentation(
+            "Merge operation for Fog VDBs.\n\n"
+            "None:\n"
+            "   Add grids to the output of the SOP without merging them.\n\n"
```

Comment:
Weird to have two meanings of Add here.
"Passes the grids to the output without affecting them"
perhaps?
"Leaves input fog grids unchanged"